tests/phpunit/CRM/Multicompanyaccounting/Hook/ValidateForm/FinancialTypeAccountTest.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/**
+ * @group headless
+ */
+class CRM_Multicompanyaccounting_Hook_ValidateForm_FinancialTypeAccountTest extends BaseHeadlessTest {
+
+  private $form;
+
+  private $financialTypeId;
+
+  private $donationFinancialAccountId;
+
+  private $accountsReceivableFinancialAccountId;
+
+  public function setUp() {
+    $this->form = new CRM_Financial_Form_FinancialTypeAccount();
+
+    $this->financialTypeId = civicrm_api3('FinancialType', 'create', [
+      'sequential' => 1,
+      'name' => "Test",
+    ])['id'];
+
+    $this->donationFinancialAccountId = civicrm_api3('FinancialAccount', 'getvalue', [
+      'return' => "id",
+      'name' => "Donation",
+    ]);
+
+    $this->accountsReceivableFinancialAccountId = civicrm_api3('FinancialAccount', 'getvalue', [
+      'return' => "id",
+      'name' => "Accounts Receivable",
+    ]);
+
+    $firstOwnerOrgId = $this->createCompany(1)['contact_id'];
+    $this->updateFinancialAccountOwner('Donation', $firstOwnerOrgId);
+
+    $this->updateFinancialAccountOwner('Accounts Receivable', $firstOwnerOrgId);
+    $this->updateFinancialAccountOwner('Banking Fees', $firstOwnerOrgId);
+    $this->updateFinancialAccountOwner('Premiums', $firstOwnerOrgId);
+    $this->updateFinancialAccountOwner(strtolower("Test"), $firstOwnerOrgId);
+  }
+
+  /**
+   * This function used to test the case where
+   * we have only one assigned financial account type
+   * since CiviCRM by default assign 4 account for the created financial type
+   */
+  private function deleteAllFinancialTypeAccountsExceptOne() {
+    \Civi\Api4\EntityFinancialAccount::delete()
+      ->addWhere('entity_table', '=', 'civicrm_financial_type')
+      ->addWhere('entity_id', '=', $this->financialTypeId)
+      ->addWhere('financial_account_id', '!=', $this->accountsReceivableFinancialAccountId)
+      ->execute();
+  }
+
+  private function deleteAllFinancialTypeAccounts() {
+    \Civi\Api4\EntityFinancialAccount::delete()
+      ->addWhere('entity_table', '=', 'civicrm_financial_type')
+      ->addWhere('entity_id', '=', $this->financialTypeId)
+      ->execute();
+  }
+
+  public function testUpdatingFinancialAccountWithOwnerAccountMatchesTheAssignedOwnerAccountsWillPassValidation() {
```

Comment:
sometimes you use `Updating` sometimes `Update` same with `Add`, try to be consistent with naming, I recommend to remove the `ing` from all of them.